src/main/java/org/jvnet/hudson/update_center/Plugin.java

```diff
@@ -460,7 +465,35 @@ public JSONObject toJSON() throws Exception {
 
         json.put("wiki", "https://plugins.jenkins.io/" + artifactId);
 
-        json.put("labels", getLabels());
+        GitHubSource gh = GitHubSource.getInstance();
+        ArrayList<String> labels = new ArrayList<String>();
+        labels.addAll(Arrays.asList(getLabels()));
+
+        if (scm != null && scm.contains("https://github.com/")) {
+            String[] parts = scm.replaceFirst("https://github.com/", "").split("/");
+            if (parts.length >= 2) {
+                labels.addAll(
+                    Arrays.asList(
+                        gh.getTopics(parts[0], parts[1]).toArray(new String[0])
+                    )
+                );
+            }
+        }
+
+        if (!labels.isEmpty()) {
+            HashSet<String> allowedLabels = new HashSet<String>();
+
+            for (String label : labels) {
+                // Everything starting with jenkins- is allowed
+                if (label.startsWith("jenkins-")) {
+                    allowedLabels.add(label.replaceFirst("jenkins-", ""));
```

Comment:
Either we have a whitelist, or we don't. Which is it?

I expect that the most common label matching this will be `jenkins-plugin`, which is really really useless.

Reply:
Yea reading it again it shouldn't be an else. It should update the string then check allowed. Will fix